app/src/androidTest/java/net/skyscanner/backpack/compose/dialog/BpkDialogTest.kt

```diff
@@ -0,0 +1,175 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.dialog
+
+import android.view.ViewGroup
+import android.view.ViewParent
+import android.widget.FrameLayout
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.isDialog
+import androidx.compose.ui.test.junit4.AndroidComposeTestRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.ActivityTestRule
+import net.skyscanner.backpack.BpkSnapshotTest
+import net.skyscanner.backpack.BpkTestVariant
+import net.skyscanner.backpack.demo.R
+import net.skyscanner.backpack.demo.compose.BackpackPreview
+import net.skyscanner.backpack.demo.compose.DestructiveDialogExample
+import net.skyscanner.backpack.demo.compose.NoIconDialogExample
+import net.skyscanner.backpack.demo.compose.SuccessOneButtonDialogExample
+import net.skyscanner.backpack.demo.compose.SuccessThreeButtonsDialogExample
+import net.skyscanner.backpack.demo.compose.SuccessTwoButtonsDialogExample
+import net.skyscanner.backpack.demo.compose.WarningDialogExample
+import org.junit.Assume
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.lang.reflect.Field
+
+@RunWith(AndroidJUnit4::class)
+class BpkDialogTest : BpkSnapshotTest() {
+
+  @get:Rule
+  var activityRule = ActivityTestRule(AppCompatActivity::class.java, true, false)
+
+  @get:Rule
+  val composeTestRule = AndroidComposeTestRule(activityRule) { it.activity }
+
+  @Before
+  fun setup() {
+    setDimensions(height = 600, width = 420)
+  }
+
+  @Test
+  fun successOneButton() = record {
+    SuccessOneButtonDialogExample()
+  }
+
+  @Test
+  fun successTwoButtons() {
+    assumeVariant(BpkTestVariant.Default, BpkTestVariant.DarkMode)
+    record {
+      SuccessTwoButtonsDialogExample()
+    }
+  }
+
+  @Test
+  fun successThreeButtons() {
+    assumeVariant(BpkTestVariant.Default, BpkTestVariant.DarkMode)
+    record {
+      SuccessThreeButtonsDialogExample()
+    }
+  }
+
+  @Test
+  fun warning() {
+    assumeVariant(BpkTestVariant.Default, BpkTestVariant.DarkMode)
+    record {
+      WarningDialogExample()
+    }
+  }
+
+  @Test
+  fun destructive() {
+    assumeVariant(BpkTestVariant.Default, BpkTestVariant.DarkMode)
+    record {
+      DestructiveDialogExample()
+    }
+  }
+
+  @Test
+  fun noIcon() {
+    assumeVariant(BpkTestVariant.Default, BpkTestVariant.DarkMode)
+    record {
+      NoIconDialogExample()
+    }
+  }
+
+  private fun record(content: @Composable () -> Unit) {
```

Comment:
as dialogs are in a separate window we need to run the screenshot tests slightly different